src/main/java/seedu/address/logic/commands/EditConcertCommand.java

```diff
@@ -0,0 +1,210 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ADDRESS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_CONCERTS;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.commons.Address;
+import seedu.address.model.commons.Name;
+import seedu.address.model.concert.Concert;
+import seedu.address.model.concert.ConcertContact;
+import seedu.address.model.concert.ConcertDate;
+
+/**
+ *
+ */
+public class EditConcertCommand extends Command {
+
+    public static final String COMMAND_WORD = "editc";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the concert indentified "
+            + "by the index number used in the displayed concert list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_NAME + "NAME" + "] "
+            + "[" + PREFIX_ADDRESS + "ADDRESS" + "] "
+            + "[" + PREFIX_DATE + "DATE" + "]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_NAME + "Coachella"
+            + PREFIX_ADDRESS + "1 Stadium drive"
+            + PREFIX_DATE + "2024-10-11 2200";
+
```

Comment:
also for this one there should be space before the prefix_address and prefix_date